src/client.js

```diff
@@ -3,8 +3,26 @@ import React from 'react';
 /*eslint-enable no-unused-vars*/
 import { render } from 'react-dom';
 import { Router, browserHistory } from 'react-router';
+import { IntlProvider, addLocaleData } from 'react-intl';
 import routes from './routes.js';
 import ga from 'react-ga';
+var locales = require('../public/locales.json');
+
+function createElement(Component, props) {
+  var locale = window.location.pathname.split("/")[1];
+  var ReactIntlLocaleData = window.ReactIntlLocaleData;
+  var messages = locales[locale];
+
+  Object.keys(ReactIntlLocaleData).forEach((lang) => {
+    addLocaleData(ReactIntlLocaleData[lang]);
+  });
+
+  return (
+    <IntlProvider locale={locale} messages={messages}>
```

Comment:
This is just wrapping whatever component is requested with the l10n context. Component is a pretty generic thing, and we don't really care what it is, just that we need to wrap it and pass along it's props.